### ISSUE TITLE
City demands banned luxury resource

### ISSUE THREAD

**Issue by UndeadDevel** [LoneGazebo/Community-Patch-DLL#5031]
_1. **Mod version** (i.e Date - 4/23):_
11-30-18

_2. **Mod list** (if using Vox Populi only, leave blank):_

VP (EUI)
City-States Leader for VP (v8)
List Enhancements for VP (v4.2)
Trade Opportunities for VP (v8.1)
Reforestation (v8) (slightly modified by me)
Civ-linked GG/GA (v11)
Enhanced Naval Warfare for VP (v1.6)
Enlightenment Era (VP) (v1.2)
Faster Aircraft Animations (v3)
Capture Great People (I neutered this one because of compatibility issues)
Espionage Mapping (v7) (modified by me)
Religious Settlers (v1)
Improved Air Recon (v1)
Missiles No Supply (v1)
More Unique Components for VP (v38)
Promotions Flags EUI (v1.1) (slightly modified by me)
Subs Ignore Borders (v3)
Subs Immune Under Ice (v2)
Destination (v4)
Filters in Trade Route Overview (VP) (v1.3) (slightly modified by me)
Improved City View (VP EUI) (v11.0)
Promotion Tree (v17)
Upgrade Tree (v9)
VP Promo Icons (v1.9)

_3. **Error description**:_
Cities can demand a luxury that has been banned by the World Council. This has been noticed by others as well (see [here](https://forums.civfanatics.com/threads/quick-questions-quick-answers.573489/page-204#post-15261484)), so I don't think it's a mod conflict.

A banned luxury shouldn't be used for anything as it's been designated as "immoral, endangered or otherwise inappropriate for use by the general population. [...]" as per the in-game description, so Cities shouldn't be demanding it for WLTKD. It also breaks the intention of the banning mechanism, which is to remove any benefit people gain from that resource, but one can nevertheless acquire it through means like conquest to get WLTKD and thus receive benefit.

If, on the other hand, this is intentional, because "sometimes people want illegal stuff" or something, then it should be possible to trade the resource (black market) and it should give happiness (the people wanted it, after all) but you can see how this doesn't make any sense so really the fix should be that people don't demand banned luxuries.

_4. **Steps to reproduce** (optional):_
play game
ban luxury/luxuries
keep pressing "Next Turn"

---------------------------
Supporting information:
Please note that you can attach .zip files by dragging-and-dropping them. If possible, zip up all supporting data and post that way.

1. Log files (always attach your Logs folder, located at My Documents/My Games/Sid Meier's Civilization 5. Make sure you have enabled logging before experiencing an error! Go here to find out how: http://forums.civfanatics.com/showthread.php?t=487482):

2. Save game (always attach a save that was made a turn before the error; located at My Documents/My Games/Sid Meier's Civilization 5/ModdedSaves):

3. CvMiniDump.dmp file (attach if experiencing a game crash. Located at Program Files/Steam/steamapps/common/Sid Meier's Civilization V):

4. Screenshots (optional):

**Comment by TechpriestEnginseer** [comment]
afaik when it runs out of available luxury pool (as in you already have them atm) to draw from then it'll draw from banned luxuries.

**Comment by UndeadDevel** [comment]
That doesn't make sense to me, can you clarify?

I don't have all other luxuries and there are several of my cities demanding luxuries, which I could get through CS alliance or conquest, that are not banned.

**Comment by TechpriestEnginseer** [comment]
A city has its own separate luxury pool to draw from.

Let's say Amber is banned, but Jade, Gold, and Silver isn't. You own a Gold luxury.

The same city drew Jade 50 turns ago and just got out of a WLTKD for Silver. The likelihood is that they want Gold, but since you already own a Gold luxury it has no other luxury to draw from except a banned luxury(which suddenly the AI prompts we should repeal the ban on it).

Historically it's like banning alcohol in a country in the early 20th century and then watching cities hunger for it after having everything they have besides alcohol.

**Comment by UndeadDevel** [comment]
I see. I doubt that this is what happened here since I only just entered Industrial and there are definitely resources I've never had before, like Porcelain, which isn't banned but can be gotten through CS alliance, so that should then still be in the pool and should be preferred over the banned resource.

Furthermore, no matter how this happened here exactly, if what you describe is actually how it works then the mechanism should be changed such that, as per your example, the city demands Jade again (or any of the other luxuries that were drawn before and I don't have).
After all, in your example with alcohol, people still drank alcohol because there was a black market for it, just as there is with all sorts of drugs right now, which is a form of trade, as well. 
Also, you are comparing a national ban on a substance with a global one, which means you have to consider the difference in position now: the nation is now the World Council and the local alcohol mafia is now the nation/Civ. Then you have the following dynamic: assuming that the Civ who has some banned luxuries still wants to trade them (which will always be true; why wouldn't they?) it means that if I, as a nation/Civ, want my people to have a luxury, I will simply make a trade with the guy who is selling it. This, however, is not possible in the game, which demonstrates how the historicity aspect breaks down.
Lastly, the AI could be voting to repeal a ban for all sorts of reasons, like pleasing another AI, making voting deals, wanting more luxuries for happiness, wanting a shot at the monopoly etc. etc.
I don't think it makes much sense to keep incoherent behavior in the game just so the AI may vote on something that it can vote on for many other reasons.

TL;DR: I filed this as a bug report and not a forum post because there is no logical consistency in a city demanding a resource I cannot trade. If you want to keep the demanding behavior and ban at the same time, make it so that the resource can still be traded but doesn't give any happiness or monopoly bonus.

**Comment by TechpriestEnginseer** [comment]
If you've had WLTKD from acquiring porcelain before, then it won't draw from the pool(because you recently acquired it). **Either way, it's something that isn't that game-breaking.** You can acquire a banned luxury through city conquest or from a city-state alliance. Or you could repeal the ban. The ban on trade was made to help the AI(not choose a dead luxury). As what usually happened is that the player will start giving out free banned luxuries to everybody and everybody suddenly wants to repeal the ban while you get free diplomacy bonus as well.
